### PR TITLE
Implement API for converting Picture to BufferedImage

### LINKED ISSUE #87: Convert `Picture` to `BufferedImage`
Create a stable Java2D specific method to convert a `Picture` to a `BufferedImage`. This will allow fun pixel art in conjunction with #85. This should be a new kind of effect type class, I think.

We already have a `ToPicture` type class. Perhaps `FromPicture` is the natural counterpart here? Not sure they would be symmetric though---rendering a picture is an effect but converting to a picture is not.

### PR DESCRIPTION
**Pull request by jCabala**
closes #87

### CODE REVIEW

File: core/shared/src/main/scala/doodle/effect/BufferedImageConverter.scala
Comment (noelwelsh) [inline core/shared/src/main/scala/doodle/effect/BufferedImageConverter.scala:1]: Does this compile on the Javascript platform? E.g. does `coreJS / fastOptJS` succeed?

I suspect the `BufferedImage` type doesn't exist on JS, and hence it won't compile. This mean either:

1. this type needs to become generic in the bitmap type or
2. this type needs to migrate to live only on the JVM platform (and possibly only on the Java2D backend.)

**Review by noelwelsh — CHANGES_REQUESTED** [review]
I think there are few issues that need to be looked at before this is ready to go.

File: core/jvm/src/main/scala/doodle/syntax/BufferedImageConverterSyntax.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        picture: Picture[Alg, A]
+    ) {
+      def apply[Frame](frame: Frame)(implicit
+          w: BufferedImageConverter[Alg, Frame, Fmt],
+          r: IORuntime
+      ): IO[(A, BufferedImage)] =
+        w.bufferedImage(frame, picture)
+    }
+
+    def bufferedImage[Fmt <: Format] =
```
[the remark this answers is not in the document]
Comment (jCabala) [reply core/jvm/src/main/scala/doodle/syntax/BufferedImageConverterSyntax.scala:52]: I just added the format becuase the `renderBufferedImage()` requires a `makeImage()` function which defined for each `Java2d[Format]Writer` object
image
Should I just create a separate object for the `BufferedImageConverter` with its own `makeImage()` function and make this the only implicit `BufferedImageConverter`?

File: core/shared/src/main/scala/doodle/effect/BufferedImageConverter.scala
Comment (jCabala) [reply core/shared/src/main/scala/doodle/effect/BufferedImageConverter.scala:1]: You are probably right. I was just following what `Base64` does. I will move it to the Java2D specific directory.

**Pull request merged**